File: math/catalan/JavaScript/catalan.js
/**
 * Catalan Number Generator
 *
 * @author Osipov Sergei 
 * @license MIT
 *
 * @param {Number} n input number
 *
 * @example
 * // const index = 3;
 * // console.log(`Catalan number #${index} is ${catalanNumber(index)}`);
 *
 * @returns {Number}
 */
const catalanNumber = (n) => {
	// unsigned int
	if ((n ^ 0) !== n || n <= 0) {
		return 0;
	}
	if (n === 1) {
		return 1;
	}
	return (4*n-2)/(n+1)*catalanNumber(n-1);
}

module.exports = catalanNumber;

File: math/catalan/C++/catalan.cpp
#include <iostream>

unsigned int catalan(unsigned int n)
{
    if (n <= 1)
    {
        return 1;
    }
 
    unsigned int res{ 0 };
    for (unsigned int i{ 0 }; i < n; ++i)
    {
        res += catalan(i) * catalan(n - i - 1);
    }
    return res;
}

int main()
{
    for (unsigned int i{ 0 }; i < 10; ++i)
    {
        std::cout << catalan(i) << std::endl;
    }
}